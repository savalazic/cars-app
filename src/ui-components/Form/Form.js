// @flow
import React, { type Element } from 'react';

import './Form.css';

import Button from '../Button';

type Props = {
  onSubmit: () => void,
  submitText: string,
  children: Element<*>,
};

const Form = ({ onSubmit, submitText, children }: Props) => (
  <form onSubmit={onSubmit}>
    {children}
    <Button type="submit" text={submitText} />
  </form>
);

export default Form;
