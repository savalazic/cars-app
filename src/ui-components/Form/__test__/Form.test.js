import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';

import Form from '../Form';

describe('Form component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Form />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(shallow(<Form />).exists(<form />)).toBe(true);
  });

  it('renders a submit button', () => {
    expect(mount(<Form />).find('button').length).toEqual(1);
  });
});
