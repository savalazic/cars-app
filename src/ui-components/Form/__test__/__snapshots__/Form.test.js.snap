// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders correctly 1`] = `
<form
  onSubmit={undefined}
>
  <button
    disabled={undefined}
  />
</form>
`;
